internal/clients/ec2/ec2_client.go

```diff
@@ -120,3 +120,34 @@ func (c *Client) ListInstanceTypes() ([]types.InstanceTypeInfo, error) {
 
 	return resp.InstanceTypes, nil
 }
+
+func (c *Client) RunInstances(ctx context.Context, amount int32, instanceType types.InstanceType, imageId string, keyName string) ([]*string, error) {
+	log.Trace().Msg("Run AWS EC2 instance")
```

Comment:
Thanks for using trace level, I feel like we already have too many debug messages. It would be useful to log also via `Msgf` and add all the arguments.

One field is extra important that is the Instance id returned by Amazon, I think this is worth logging at info level in a separate logging field like `instance_id` so we can search for that easily.

Reply:
I am parsing the response from Amazon, wondering if it is necessary, if we want to display it, or just log it.

Reply:
I would only log this. For each instance ID I would create a new log entry with "Created new instance via reservation XXX" (reservation id returned from AWS). And each log entry would have a field named "instance_id" with the instance id included. This way we can easily search the logs and find the whole transaction responsible for doing the request in case something went wrong.